From f0702d6e631bb912a230c081463bb51a0dde1bff Mon Sep 17 00:00:00 2001
From: Jon TURNEY <jon.turney@dronecode.org.uk>
Date: Mon, 12 May 2014 15:38:26 +0100
Subject: [PATCH 1001/1003] appleglx: Improve error reporting if
 CGLChoosePixelFormat() didn't find any matching pixel formats.

Signed-off-by: Jon TURNEY <jon.turney@dronecode.org.uk>
Reviewed-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit 002a3a74273b81dfb226e1c3f0a8c18525ed0af4)
---
 src/glx/apple/apple_visual.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/glx/apple/apple_visual.c b/src/glx/apple/apple_visual.c
index 238c248..c6ede51 100644
--- a/src/glx/apple/apple_visual.c
+++ b/src/glx/apple/apple_visual.c
@@ -167,4 +167,9 @@ apple_visual_create_pfobj(CGLPixelFormatObj * pfobj, const struct glx_config * m
       fprintf(stderr, "error: %s\n", apple_cgl.error_string(error));
       abort();
    }
+
+   if (!*pfobj) {
+      fprintf(stderr, "No matching pixelformats found, perhaps try using LIBGL_ALLOW_SOFTWARE\n");
+      abort();
+   }
 }
-- 
1.9.2 (Apple Git-49)

